streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -637,6 +641,33 @@ public void shouldNotThrowTaskMigratedExceptionDuringRestoreForChangelogTopicWhe
         assertThat(callback.restored.size(), equalTo(10));
     }
 
+    @Test
+    public void shouldReinitializeStateStoresWhenNoCheckpointFoundAndEOSEnabled() {
```

Comment:
Should we have a reverse test too, that check that a task is not re-initialized if a checkpoint file is found?

Reply:
ack good point I'll add a test for that.

Reply:
I'll do a follow-on PR for the additional test